Makefile

```diff
@@ -429,6 +429,34 @@ build-archive:
 release-unix: clean full build-archive
 	@if [ -f e/Makefile ]; then $(MAKE) -C e release; fi
 
+# Set MacOS/Darwin cert vars for signing/notarizing
+# TSH_SKELETON is a directory name relative to build.assets/macos/
+ENVIRONMENT_NAME ?= promote
```

Comment:
Can all this new stuff be moved into a separate .mk file (maybe `darwin.mk` or something) and included here?

Reply:
I was considering that, but wasn't sure where to put it. I could put it at the top-level, but that's a bit polluting. We do have `common.mk` there already though. I could put it in `build.assets/` but that already has a `Makefile` so it may look like it is part of that `Makefile` not the top-level one. I could create a new top-level directory `mk/` and put it there and even move the others there. But that's more churn than I want right now.

So in the absence of a winning idea, I just put it in the top-level `Makefile` :-|

Do you have a preference?

Reply:
I went with `darwin-signing.mk` as I felt that `darwin.mk` should have all the darwin stuff in it. Happy to rename it to the shorter name if you prefer.